projects/cli-files/go-cat/cmd/root.go

```diff
@@ -0,0 +1,66 @@
+package cmd
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+)
+
+func Execute() {
+	args := os.Args[1:]
+
+	for _, arg := range args {
+		err := checkArgs(arg)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return
+		}
+		fileLines, err := readFileLines(arg)
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+		}
+		for _, fileLine := range fileLines {
+			fmt.Print(fileLine)
+		}
+	}
+}
+
+func checkArgs(arg string) error {
+	if arg == "" {
+		return fmt.Errorf("go-cat: no file specified")
+	}
+	fileInfo, err := os.Stat(arg)
```

Comment:
Have a read about [TOCTOU](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use) :) This is fine, it's ok to do this kind of check to be able to give a more specific error, but it's possible e.g. that the file is deleted, or replaced by a directory with the same name _after_ you do the check but _before_ you try to read it. We _generally_ avoid doing "check if this is likely to work" checks, and favour just trying to do the thing and reporting the error if it failed, but sometimes we do the check before anyway :)

Reply:
Now I understand why you said it's more complicated xD